lib/libzfs/libzfs_sendrecv.c

```diff
@@ -2060,12 +2060,12 @@ send_prelim_records(zfs_handle_t *zhp, const char *from, int fd,
 	int err = 0;
 	char *packbuf = NULL;
 	size_t buflen = 0;
-	zio_cksum_t zc = { {0} };
+	zio_cksum_t *zc = calloc(sizeof (zio_cksum_t), 1);
```

Comment:
This seems a little heavy handed. Is this for alignment purposes? Couldn't you just use a compiler directive to align this on the stack?

Reply:
Yes, and then we get to update it every time we change SIMD alignment requirements, and god help us if someone does something clever and they're no longer mutually compatible.

The implementations are all written entirely with unaligned accesses anyway (except possibly the NEON one? I'll go boot my AArch64 testbed and see...), I'm just going to drop the alignment definition in the union, because we're basically just saying "pretty please" and wishing we had our cake and ate it too, now.

Reply:
64 byte alignment is pretty futureproof, no? It might add some padding to your stack location but it's going to be way better than allocating dynamically.

Reply:
Having checked, the NEON implementation also appears to not require the alignment at all, so I think I prefer suggesting people add alignment if there are places they feel the difference is a significant win, rather than pretending it's aligned everywhere and then having to manually force things we're munging into `fletcher_4_ctx_t *` to make that true.

(That said, UBsan also screams murder on AArch64 about the LZ4 implementation unrelated to these changes, so that's a different problem...)

Reply:
So for an infrequent enough access, I don't think it's a significant win much anymore on newer CPUs.  For AARCH64, particularly on "little" cores, aligned access is still quite a bit faster than unaligned.  I get the impression this checksum is only written back to memory after summing a block's worth of bytes, so it's probably not going to be significant either way.

Now, on PowerPC, particularly the older, pre-power7 variants, alignment is a requirement or it takes two loads and a permute from a permutation vector returned from vec_lvsl.  I'm not sure how far back OpenZFS's PowerPC support goes back and if it needs the VSX extensions or supports the less VMX/altivec ones.